src/cow-react/common/pure/CancelButton/index.tsx

```diff
@@ -0,0 +1,33 @@
+import { LinkStyledButton } from 'theme'
+import { CancellationModal } from './CancellationModal'
+
+// TODO: add for each type when implemented
+export type CancelButtonType = 'soft'
+
+export type CancelButtonContentProps = {
+  orderId: string
+  summary: string | undefined
+  chainId: number
+  error: string | null
+  onCancelButtonClick: () => void
+  onDismiss: () => void
+  showCancelModal: boolean
+  type: CancelButtonType
+  softCancellationContext?: SoftCancellationContext
+}
+
+export type SoftCancellationContext = {
+  cancelOrder: () => void
+  isWaitingSignature: boolean
+}
+
+export function CancelButtonContent(props: CancelButtonContentProps) {
+  const { showCancelModal, onCancelButtonClick, ...rest } = props
+
+  return (
+    <>
+      <LinkStyledButton onClick={onCancelButtonClick}>Cancel order</LinkStyledButton>{' '}
+      {showCancelModal && <CancellationModal {...rest} />}
```

Comment:
Does it make sense to make a button have the modal? 
Or should it be that the button muttates a piece of state that shows the modal. So we only have one modal.

Reply:
This is one of the things that bother me, and this is the original code.
You think the consumers should add these components independently?